records_mover/session.py

```diff
@@ -109,15 +118,28 @@ def __init__(self,
         if default_aws_creds_name is PleaseInfer.token:
             default_aws_creds_name = _infer_default_aws_creds_name(session_type)
 
+        if default_gcp_creds_name is PleaseInfer.token:
+            default_gcp_creds_name = _infer_default_gcp_creds_name(session_type)
+
         self._default_db_creds_name = default_db_creds_name
         self._default_aws_creds_name = default_aws_creds_name
+        self._default_gcp_creds_name = default_gcp_creds_name
         self._scratch_s3_url = scratch_s3_url
         self.creds = creds
-        url_resolver_kwargs = {}
+        self._records: Optional[Records] = None
+
+    @property
+    def url_resolver(self) -> UrlResolver:
+        # TODO: Make cache
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker